snippets/csharp/System.Linq/Enumerable/AggregateTSource/enumerable.cs

```diff
@@ -109,32 +106,29 @@ public static void AllEx()
         static class All2
         {
             // <Snippet129>
-            class Pet
+            class Pet(string name, int age)

```

Comment:
Why specifically adopt a primary constructor in this example? The example could simply update the nullability annotation for `Name` or enforce immutability using `init` (and possibly `required`) instead of `set` if that's the end goal.

Reply:
I would prefer `record` or `record struct` types for this and other data types defined in these samples. The use (a data container) is the main scenario for `record` types. In the context of the sample, I don't have a strong preference for `record class` vs. `record struct` or enforcing immutability.